### PR TITLE
Improving OTP generation

### PR DESCRIPTION
**Pull request by bohdansec**
I'd like to propose a minor improvement aimed at enhancing the cryptographic strength and expanding the capabilities of OTP generation. Specifically, this would enable the creation of OTPs that can start with zero and allow for dynamically defining the length of the OTP.

Additionally, if the functionality is sufficiently generic and does not include business logic specific to a single service, consider the possibility of extracting such code into a common library that can be utilized by both services. This will ensure code reusability and prevent duplication. For microservices, this often pertains to utility functions, such as OTP generation.

I also recommend increasing the minimum length of the length parameter for OTPs from 4 to 6-8 characters. You might also consider storing the value of the length parameter in the configuration.

### CODE REVIEW

**Comment by alinkedd** [comment]
Implements https://github.com/diia-open-source/be-user-service/issues/63

**Comment by includesec-erik** [comment]
Strongly support @bohdansec 's proposal in comments to move from 4 chars to 6 on the OTP token.

Also @bohdansec , if you'd like to collab on doing a security review of this code hit us up codeaudit [at] IncludeSecurity.com

File: src/services/otp.ts [hunk 1]
```diff
@@ -28,7 +29,7 @@ export default class OtpService {
             throw new AccessDeniedError('Exceeded otps limits per day', undefined, ProcessCode.OtpSendAttemptsExceeded)
         }
 
-        const value: number = Math.floor(1000 + Math.random() * 9000)
+        const value: string = this.generateOtp(4)
 
         await this.notificationService.sendSms(phoneNumber, SmsTemplateCode.Otp, user, headers, value.toString())
```
Comment (bartcubbins) [inline src/services/otp.ts:34]: >  value.toString()

value has already become a string https://github.com/diia-open-source/be-user-service/pull/99/commits/c056296d35b3634993973321afcd7e63939c2dcf#diff-00fedd47698c4954f50bd0b9a727028195d4eb26dfaba6d07dbf153d870b142aR32

**Comment by FowiFlowi** [comment]
@bohdansec, thank you for PR! Can you please resolve conflicts and conversations so we can accept it?

**Comment by FowiFlowi** [comment]
Actually, this OTP logic has never been connected to any public services. So we can claim it as a dead code. We already removed this functionality but created a separate util to generate such strings - https://github.com/diia-open-source/be-pkg-utils/blob/main/src/random.ts

**Pull request closed without merging**